tsfc/driver.py

```diff
@@ -329,22 +415,67 @@ def compile_expression_at_points(expression, points, coordinates, interface=None
     expression = ufl_utils.split_coefficients(expression, builder.coefficient_split)
 
     # Translate to GEM
-    point_set = PointSet(points)
-    config = dict(interface=builder,
-                  ufl_cell=coordinates.ufl_domain().ufl_cell(),
-                  precision=parameters["precision"],
-                  point_set=point_set,
-                  argument_multiindices=argument_multiindices)
-    ir, = fem.compile_ufl(expression, point_sum=False, **config)
-
-    # Deal with non-scalar expressions
-    value_shape = ir.shape
-    tensor_indices = tuple(gem.Index() for s in value_shape)
-    if value_shape:
-        ir = gem.Indexed(ir, tensor_indices)
+    kernel_cfg = dict(interface=builder,
+                      ufl_cell=coordinates.ufl_domain().ufl_cell(),
+                      precision=parameters["precision"],
+                      argument_multiindices=argument_multiindices,
+                      index_cache={})
+
+    if all(isinstance(dual, PointEvaluation) for dual in to_element.dual_basis()):
```

Comment:
So the 'then' branch if basically equivalent to the old code path, and then the 'else' branch deals with the general case...

Reply:
Yep.